src/liboslexec/llvm_gen.cpp

```diff
@@ -360,23 +361,27 @@ LLVMGEN (llvm_gen_printf)
 
                     llvm::Value* loaded = nullptr;
                     if (rop.use_optix() && simpletype.basetype == TypeDesc::STRING) {
-                        // TODO: make this work with arrind
-                        // In the OptiX case, we use the device-side string.
-                        loaded = rop.llvm_load_device_string (sym);
+                        // In the OptiX case, we register each string separately.
+                        if (simpletype.arraylen >= 1) {
+                            ustring name = ustring::format("%s[%d]", sym.name(), a);
+                            Symbol lsym(name, TypeDesc::TypeString, sym.symtype());
+                            lsym.data(&((ustring*)sym.data())[a]);
+                            loaded = rop.llvm_load_device_string (lsym);
+                        } else {
+                            loaded = rop.llvm_load_device_string (sym);
+                        }
+                        optix_size += sizeof(uint64_t);
                     }
                     else {
                         loaded = rop.llvm_load_value (sym, 0, arrind, c);
-                    }
-
-                    if (simpletype.basetype == TypeDesc::FLOAT) {
-                        // C varargs convention upconverts float->double.
-                        loaded = rop.ll.op_float_to_double(loaded);
-                    }
 
-                    if (simpletype.basetype == TypeDesc::INT && rop.use_optix()) {
```

Comment:
Are you losing the ability to do ints properly here? Why did we used to convert to longlong, but now we just leave it as int?

Reply:
The ints in va_args Cuda is expecting is int32.
Handling INT64 could be added, but does OSL have 64-bit integer types?

Reply:
No, no int64 in osl. OK, so it was broken before and now it's ok? Good.